linux/shim/io.go

```diff
@@ -15,8 +15,9 @@ import (
 )
 
 func copyConsole(ctx context.Context, console console.Console, stdin, stdout, stderr string, wg, cwg *sync.WaitGroup) error {
+	fctx := context.Background()
 	if stdin != "" {
-		in, err := fifo.OpenFifo(ctx, stdin, syscall.O_RDONLY, 0)
+		in, err := fifo.OpenFifo(fctx, stdin, syscall.O_RDONLY, 0)
```

Comment:
@crosbymichael That only happens if the select is not reached before the context is cancelled, what doesn't seem likely if these fifos block. You can make sure that `Openfifo` never returns before select is reached. Or do an extra check in done

```
case <-ctx.Done():
   select {
     case <-f.opened:
     default:
  }
```